eth_verifier/lib/bn254/BN254.sol

```diff
@@ -146,7 +146,7 @@ library BN254 {
 
     /// @return r the product of a point on G1 and a scalar, i.e.
     /// p == p.mul(1) and p.add(p) == p.mul(2) for all points p.
-    function scalarMul(G1Point memory p, uint256 s) internal view returns (G1Point memory r) {
+    function scalarMul(G1Point memory p, uint256 s) public view returns (G1Point memory r) {
```

Comment:
Try changing function visibility back to `internal` to see if we can lower the gas consumption.

Reply:
done